Automattic/jetpack
# My Jetpack Welcome Flow: Show default recommendations upfront first, then offer optional survey

This PR updates the My Jetpack Welcome Flow so that after connection, some default recommendations will load upfront first to the user (as opposed to showing the survey form first). Then the user has the option to get customized results by selecting to submit the survey form.

This updated flow will be run within an Explat experiment as the "treatment" variation.

Fixes: https://github.com/Automattic/jetpack-marketing/issues/1016




### Screencast of the "treatment" variation of the Welcome Flow in action (this PR):

https://github.com/user-attachments/assets/fa52eb68-eb2e-483c-87a2-13f0a861f91f


## Proposed changes:

Implemented 2 different welcome flow experiences:
* Control: Upon site connection, the user is presented with a survey asking what their site goals are (with checkboxes). Upon submission of the survey, the user given product recommendations based on their site goals. (Same as before: No changes.)
* **Treatment:** Upon site connection, the user is presented with product recommendations first (based on site defaults), with the option to better customize their recommendations by submitting the survey with their site goals.
    * Updated some copy & added 2 links below the recommended products: 1 for submitting site goals for customized results, and the other to view all site plans (pricing page).

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-bvL-p2
Figma designs: TzLqUupJf9CmSYsAplk04n-fi-4515_1135

Explat experiment: 22033-explat-experiment

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






Prior to testing, go to the Explat experiment page: 22033-explat-experiment - Manually assign the "treatment" variation to your browser using the bookmarklet:
- Go to the Audience section on the experiment overview page ⤴️ and hover the “Bookmarklet” link of the variation you’d like to be assigned to. Drag the link into the bookmarks bar of your browser. Then on the page where your are going to start testing the experiment, click the variation bookmark. You should get a confirmation that the variation was assigned successfully. Then test the experiment. For more information on testing experiments, see: PCYsg-SwK-p2

* Checkout this branch via the Jetpack Beta plugin (on a new JN site) or your local dev environment
    - If using local dev environment (monorepo docker), update `projects/packages/my-jetpack/src/class-initializer.php`, line 457, Add: `return true` in the `is_jetpack_user_new()` function.
* Go to the My Jetpack page.
* In the top welcome banner, click the "Connect Jetpack in one click" button.
* Immediately after connection, verify that the Recommendations loading banner displays briefly (See screencast above). It says, "Finding the best Jetpack tools" with a loading spinner, and some additional descriptive text.
* Verify that (3) product recommendations appear. (Take note which products appear, and their order)
* Verify that these initial Recommendations follow this design: TzLqUupJf9CmSYsAplk04n-fi-4515_2191
    - Verify the Heading is: "Recommended for your site"
    - Verify there are 2 links below the recommended products, with copy, "Find your perfect match by letting us know what you’re looking for!" and "Explore all Jetpack plans".
    -  Verify the context menu items are: "Customize recommendations", and "Close".
* Next click either the link, ""Find your perfect match by letting us know what you’re looking for!", or in the context menu "Customize recommendations" link, to be shown the survey.
* Check one of the site goals and click "See solutions"
* Verify you are shown new results/recommendations that are different than the first default ones.
* Verify that these tailored recommendations follow this different design: TzLqUupJf9CmSYsAplk04n-fi-4515_2794
    - Verify the Heading is: "Our recommendations for you"
    - Verify there are 2 links below the recommended products, with copy, "Start over? Analyze again for fresh recommendations!" and "Explore all Jetpack plans".
    -  Verify the context menu items are: "Redo", and "Dismiss".
----------
- To start over fresh and test the flow again (I.e- To test the flow with the "control" variation assigned):
- SSH into your site (For jurassic Ninja, use the SSH credentials on the WordPress WP-Admin dashboard page.)
- Open the WP-CLI (`wp` command for Jurassic Ninja, or `wpsh` command if you're testing locally).
- Reset all the necessary options:
    - run command: `return Jetpack_Options::delete_option( 'recommendations_evaluation' );`
    - run command: `return Jetpack_Options::delete_option( 'dismissed_welcome_banner' );`
    - run command: `return Jetpack_Options::update_option( 'recommendations_first_run', true );`
- Then on My Jepack page, scroll down to the "Connection" section, click "Manage" link and click "Disconnect"
- When you reload My Jetpack page, and you should be back to the beginning of the flow again (the Welcome Flow async connection banner should be showing).
- Assign the "control" experiment by clicking the control bookmark you added into your bookmarks bar (in first steps).
- Test the flow again and verify that after connection, the evaluation survey is presented upfront first (instead of default recommendations). This flow should be the same, unchanged as it always has been prior to this PR.

### More screen captures:

Control    |  Treatment
:---------:|:-----------:
![Screen Capture on 2024-09-26 at 18-17-41](https://github.com/user-attachments/assets/a4713788-6a7a-45c9-adf0-56dbb2e176da)  |  ![Screen Capture on 2024-09-26 at 18-10-38](https://github.com/user-attachments/assets/b878adb5-e57c-4fb7-b172-e48dd7c26a10)
